pywt/_extensions/_dwt.pyx

```diff
@@ -98,9 +98,9 @@ cpdef dwt_axis(np.ndarray data, Wavelet wavelet, MODE mode, unsigned int axis=0)
     return (cA, cD)
 
 
-cpdef idwt_single(data_t[::1] cA, data_t[::1] cD, Wavelet wavelet, MODE mode):
-    cdef data_t[::1] rec
+cpdef idwt_single(np.ndarray cA, np.ndarray cD, Wavelet wavelet, MODE mode):
```

Comment:
I guess that if the types are already being checked then I can remove the `cA.dtype != cD.dtype` check I had added.

I will check later this evening or weekend if changing just the inputs back to memory views has any effect.